.github/workflows/docs-build-push.yml

```diff
@@ -0,0 +1,36 @@
+name: Build and deploy docs
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Environment to deploy to'
+        required: true
+        default: 'preview'
+        type: choice
+        options:
+        - preview
+        - dev
+        - staging
+        - prod
+  pull_request:
+    branches:
+      - "*"
+    paths:
+      - "docs/**"
```

Comment:
Our public docs live under the `site/` directory. The `docs/` directory is just developer docs in Github.